File: c/src/Structural/Adapter/looseleaftea.c
//looseleaftea - the adaptee

#include "looseleaftea.h"

#include "stdio.h"
#include "mem.h"


looseleaftea_t * looseleaftea_new()
{
	looseleaftea_t * t;
	NEW(t);
	t->teaIsSteeped = 0;
	return t;
}

void looseleaftea_free(looseleaftea_t * llt)
{
	FREE(llt);
}

void looseleaftea_steep_tea( looseleaftea_t * t)
{
	t->teaIsSteeped = 1;
	printf("tea is steeping\n");
}

File: javascript/src/Structural/Adapter/LooseLeafTea.js
(function() {
  define([], function() {
    var LooseLeafTea;
    LooseLeafTea = (function() {
      LooseLeafTea.prototype.teaIsSteeped = null;

      function LooseLeafTea() {
        this.teaIsSteeped = false;
      }

      LooseLeafTea.prototype.steepTea = function() {
        this.teaIsSteeped = true;
        return 'tea is steeping';
      };

      return LooseLeafTea;

    })();
    return LooseLeafTea;
  });

}).call(this);
